icechunk/src/session.rs

```diff
@@ -178,23 +178,7 @@ impl Session {
     /// Deletes of non existing groups will succeed.
     pub async fn delete_group(&mut self, path: Path) -> SessionResult<()> {
         match self.get_group(&path).await {
-            Ok(parent) => {
```

Comment:
@paraseba this is the error. Both this loop and `change_set.delete_group` are deleting children. 

So if there is `/group` and `/group/subgroup`. `nodes_iter` contains both those groups. IF `/group` appears first then `change_set.delete_group` will delete both groups. Then in the second iteration `change_set.delete_group("/group/subgroup")` will add an entry to `change_set.deleted_groups`.

Reply:
(this change is not right). I'll try filtering out any nodes added in the currnet session